### PR TITLE
Test against the stable endpoint from MSC2946

### PR DESCRIPTION
**Pull request by clokep**
See matrix-org/synapse#11329 for the corresponding synapse PR. I've added this to the dendrite blacklist and also removed the old tests against `/spaces`.

Reviewable commit-by-commit.

### CODE REVIEW

File: tests/msc2946_test.go [hunk 1]
```diff
@@ -204,40 +204,7 @@ func TestClientSpacesSummary(t *testing.T) {
 	// - Rooms are returned correctly along with the custom fields `room_type`.
 	// - Events are returned correctly.
 	t.Run("query whole graph", func(t *testing.T) {
-		res := alice.MustDo(t, "GET", []string{"_matrix", "client", "unstable", "org.matrix.msc2946", "rooms", root, "spaces"}, nil)
-		must.MatchResponse(t, res, match.HTTPResponse{
```
Comment (kegsay) [inline tests/msc2946_test.go:208]: Why are we removing these checks?

**Review by kegsay — APPROVED** [review]

File: tests/msc2946_test.go (hunk 1 above)
Comment (clokep) [reply tests/msc2946_test.go:208]: It doesn't seem useful to keep them around as this endpoint is deprecated and will be removed.

**Comment by kegsay** [comment]
Happy for me to merge this?

**Comment by clokep** [comment]
> Happy for me to merge this?

No, this is waiting for corresponding Synapse changes.

**Comment by clokep** [comment]
Synapse changes got approved, so going to merge this. Thank you for the review @kegsay!

**Pull request merged**